webapp/channels/src/components/section_notice.test.tsx

```diff
@@ -0,0 +1,70 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import {fireEvent, screen} from '@testing-library/react';
+import type {ComponentProps} from 'react';
+import React from 'react';
+
+import {renderWithContext} from 'tests/react_testing_utils';
+
+import SectionNotice from './section_notice';
+
+type Props = ComponentProps<typeof SectionNotice>;
+
+function getBaseProps(): Props {
```

Comment:
If you wrote this as a function instead of making a constant `baseProps` like in other tests so that it resets the mocks between tests, that isn't actually needed. Jest does that automatically.

Reply:
This is something I am doing lately in all my tests. The main reason is that this way, doesn't matter how deep are the props or the state, I am sure the whole tree is new. So I can do:
```
const props = getBaseProps();
props.propToChange.something = 'foo'
```
Instead of
```
const props = {
    ...baseProps,
    propToChange: {
        ...baseProps.propToChange,
        something: 'foo'
    }
}
```

Maybe it is not needed in this case, but I got the "inertia" doing it, and put it here. Do you want me to get back to a constant?

Reply:
Going back to change this, I saw in the second test I do indeed that (remove the different buttons). So... I will keep it as a function.

Reply:
Oh, that's a really good idea. I haven't run into that with component props as much, but I know it happens in most tests involving Redux. I was using `mergeObjects` there, but it does still mean that you end up with 
```js
const props = mergeObjects(baseProps, {
    propToChange: {
        something: 'foo',
    },
});
```
which is wordier. That also returns an `any` which isn't great, although I do think that'd be fixable.

I'm not ready yet to go commit to rewriting everywhere that we've used a constant `baseProps` or `baseState` as a function like this, but I am sort of considering it now...